app/models/order_detail.rb

```diff
@@ -856,9 +855,17 @@ def has_completed_reservation?
     !product.is_a?(Instrument) || (reservation && (reservation.canceled_at || reservation.actual_end_at || reservation.reserve_end_at < Time.zone.now))
   end
 
+  def time_for_policy_lookup
+    if fulfilled_at
+      fulfilled_at
+    elsif reservation.try(:canceled?)
+      Time.current
```

Comment:
This will fall back to nil `nil` to and call `cheapest_price_policy(self, nil)`, so it doesn't take the default. In there, it will not find any `cheapest_price_policy(nil)`, which is what I want. This helps ensure that we don't accidentally do this again, because we should never be finding the actual price policy it's fulfilled or it's canceled with charge (which doesn't have a `fulfilled_at`, but does end up in completed).